src/variety/kuroclone.js

```diff
@@ -0,0 +1,232 @@
+			this.drawBorders();
+			this.drawChassis();
+			this.drawBoxBorders(false);
+			this.drawTarget();
+		}
+	},
+
+	//---------------------------------------------------------
+	// URLエンコード/デコード処理
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeBorder();
+			this.decodeArrowNumber16();
+		},
+		encodePzpr: function(type) {
+			this.encodeBorder();
+			this.encodeArrowNumber16();
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeAreaRoom();
+			this.decodeCellDirecQnum();
+			this.decodeCellAns();
+		},
+		encodeData: function() {
+			this.encodeAreaRoom();
+			this.encodeCellDirecQnum();
+			this.encodeCellAns();
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkSideAreaShadeCell",
+			"checkArrowNumber",
+			"checkUnitsCount",
+			"checkUnitsShape"
+		],
+
+		// From shimaguni.js
+		checkSideAreaShadeCell: function() {
+			this.checkSideAreaCell(
+				function(cell1, cell2) {
+					return cell1.isShade() && cell2.isShade();
+				},
+				false,
+				"cbShade"
+			);
+		},
+
+		// Check unit size pointed by arrow
+		checkArrowNumber: function() {
+			var bd = this.board;
+			for (var c = 0; c < bd.cell.length; c++) {
+				var cell = bd.cell[c];
+				if (!cell.isValidNum() || cell.qdir === 0) {
+					continue;
+				}
+				var pos = cell.getaddr();
+				var cell2 = pos.movedir(cell.qdir, 2).getc();
+
+				if (cell2.sblk) {
+					if (cell2.sblk.clist.length === cell.qnum) {
+						continue;
+					}
+				}
+
+				this.failcode.add("anUnitNe");
+				if (this.checkOnly) {
+					break;
+				}
+				cell.seterr(1);
+				if (cell2.sblk) {
+					cell2.sblk.clist.seterr(1);
+				}
+			}
+		},
+
+		// Check number of units is two for each room
+		checkUnitsCount: function() {
+			var rooms = this.board.roommgr.components;
+			for (var i = 0; i < rooms.length; i++) {
+				var units = this.getUnits(rooms[i]);
+				if (units.length === 2) {
+					continue;
+				}
+				this.failcode.add("bkUnitNe2");
+				if (this.checkOnly) {
+					break;
+				}
+				rooms[i].clist.seterr(1);
+			}
+		},
+
+		// Check shapes of the two units for each room
+		checkUnitsShape: function() {
+			var rooms = this.board.roommgr.components;
+			for (var i = 0; i < rooms.length; i++) {
+				var units = this.getUnits(rooms[i]);
+				// check only when the number of units is two
+				if (units.length !== 2) {
+					continue;
+				}
+				if (this.isDifferentShapeBlock(units[0], units[1])) {
+					this.failcode.add("bkDifferentShape");
+					if (this.checkOnly) {
+						break;
+					}
+					units[0].clist.seterr(1);
+					units[1].clist.seterr(1);
+				}
+			}
+		},
+
+		// get all unit component in the room
+		getUnits: function(room) {
+			var set = new Set();
+			room.clist.each(function(cell) {
+				if (cell.isShade()) {
+					set.add(cell.sblk);
+				}
+			});
+			return Array.from(set);
+		}
+	},
+
+	FailCode: {
```

Comment:
The FailCode object isn't required here, since all of these string keys are the defaults.